### PR TITLE
Take busy frames into account for peer selection

### PR DESCRIPTION
**Pull request by Raynos**
This allows a peer to favor non-busy peers.

r: @jcorbin @rf @shannili

### CODE REVIEW

**Comment by kriskowal** [comment]
:100:

**Comment by ShanniLi** [comment]
test?

**Comment by Raynos** [comment]
Working on tests.

**Comment by Raynos** [comment]
Ready for review.

I added ALL THE TESTS.

cc @jcorbin @ShanniLi

File: errors.js [hunk 1]
```diff
@@ -846,3 +846,27 @@ module.exports.logLevel = function errorLogLevel(err, codeName) {
...
+    switch (codeName) {
+        case 'Busy':
+        case 'Declined':
+            return true;
+
+        case 'BadRequest':
+        case 'Cancelled':
+        case 'NetworkError':
+        case 'ProtocolError':
+        case 'Timeout':
```
Comment (jcorbin) [inline errors.js]: `Timeout` is probably also true

File: errors.js [hunk 2]
```diff
@@ -846,3 +846,27 @@ module.exports.logLevel = function errorLogLevel(err, codeName) {
...
+        case 'UnexpectedError':
```
Comment (jcorbin) [inline errors.js]: `UnexpectedError` is probably also true

Comment (Raynos) [inline errors.js]: WHY ALL THE THINGS.

**Comment by jcorbin** [comment]
lgtm, expanding the classification can happen later

File: test/pool-of-servers.js [hunk 3]
```diff
@@ -179,12 +484,122 @@ allocCluster.test('request().send() to a pool of servers', {
...
+
+            var lower = null;
+            var upper = null;
+
+            // If its the error frame
+            if (keys[k].indexOf('oops') === 0) {
+                lower = EXPECTED_ERROR;
+                upper = EXPECTED_ERROR + 1;
+            } else {
+                lower = numExpectedReqs * 0.5;
```
Comment (ShanniLi) [inline test/pool-of-servers.js]: flappy? :)

Comment (Raynos) [inline test/pool-of-servers.js]: Nope; not really.

**Pull request merged**